truffle/src/com.oracle.truffle.api.dsl/src/com/oracle/truffle/api/dsl/Introspection.java

```diff
@@ -241,8 +241,8 @@ SpecializationInfo getSpecialization(String methodName) {
 
     private void checkVersion() {
         int version = -1;
-        Object objectVersion = data[0];
-        if (data.length > 0 && objectVersion instanceof Integer) {
+        if (data.length > 0 && data[0] instanceof Integer) {
```

Comment:
@chumer seems like `data.length` is never 0 in practice otherwise we would have seen `ArrayIndexOutOfBoundsException`s here.

Reply:
No it should not be. But the change makes it more robust. Thanks!